### PR TITLE
Make ISTIO_META_DNS_CAPTURE ISTIO_META_PROXY_XDS_VIA_AGENT as bool var

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: operator/cmd/mesh/testdata/manifest-generate/data-snapshot/profiles/preview.yaml [hunk 1]
```diff
@@ -10,9 +10,9 @@ spec:
       proxyMetadata:
         # Enable Istio agent to handle DNS requests for known hosts
         # Unknown hosts will automatically be resolved using upstream dns servers in resolv.conf
-        ISTIO_META_DNS_CAPTURE: "ALL"
```
Comment (howardjohn) [inline operator/cmd/mesh/testdata/manifest-generate/data-snapshot/profiles/preview.yaml:13]: isn't this backwards incompatible? people will probably have it set explicitly

Comment (hzxuzhonghu) [reply operator/cmd/mesh/testdata/manifest-generate/data-snapshot/profiles/preview.yaml:13]: It is from #26954, introduced recently not in 1.7

**Comment by rshriram** [comment]
how does it work now? i tried this a while back and something in yaml conversion kept failing

**Review by rshriram — APPROVED** [review]
thanks!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**